### ISSUE TITLE
convert to openvino

### ISSUE THREAD

**Issue by linhaoqi027** [ultralytics/yolov5#697]
## ❔Question

hi! I want to convert yolov5s to openvino. First it will convert to onnx, it is no problem, ans then when i convert onnx to openvino. It occurs problem.
## Additional context
![捕获](https://user-images.githubusercontent.com/46484513/89792476-2d7c9b00-db57-11ea-9651-0be147b394db.JPG)

**Comment by glenn-jocher** [comment]
Sorry buddy, can't help, no experience. If you get this to work it might be useful to write a tutorial that we could add to https://docs.ultralytics.com/yolov5 to help others.

**Comment by Aktcob** [comment]
ha, I am faced with the same problem. 

A possible way is that, the node 'Upsampling' with nearest mode can be converted to 'Upsampling' op(in ONNX) not the 'resize' op. It seemed that there is bug when 'resize' op in ONNX--->'interpolate' op in OPENVINO. And the Upsampling op in ONNX is ok.

Another way is that do not use the Upsampling op in Pytorch. You can copy the value like https://github.com/dmlc/gluon-cv/blob/master/gluoncv/model_zoo/yolo/yolo3.py#L31

**Comment by linhaoqi027** [comment]
I update the openvino to 2020R4, it is solved

**Comment by anhnktp** [comment]
@linhaoqi027 which opset value you used in converting to onnx. I'm facing the issues when convert yolov5 .pth model to openvino with opset = 10 (can't convert full the model). If I use opset = 11 or 12, I can conver full model to onnx but meet error resize operator only support for opset = 10 in openvino 2020R4.
Could you pls give your script convert to openvino ?

**Comment by linhaoqi027** [comment]
opset=10

**Comment by linhaoqi027** [comment]
I can convert to openvino

**Comment by anhnktp** [comment]
@linhaoqi027 thanks. I also use opset = 10 now, but I have trouble on final result after converting openvino model. I can output 3 features map from converted model and then apply some numpy calculation & nms step to output bounding bo (followed #343 ) . But its seems NMS step doesn't working. Could you share your script to inference of converted openvino model ? if possile, could you give me your contact ? Thank you in advance !!!
![image](https://user-images.githubusercontent.com/30823943/90867641-e9dc2980-e3bf-11ea-9359-8b2c51a031ad.png)

**Comment by samirMeziane** [comment]
@anhnktp can you tell me please if you have output with 5d shape like 1x80x80x3x7 cause i have an error when i run my transformed yolov5 openvino (outlayer ie::layout::NCDHW || outlayer ie::layout::NDHWC)

**Comment by linhaoqi027** [comment]
> @linhaoqi027 thanks. I also use opset = 10 now, but I have trouble on final result after converting openvino model. I can output 3 features map from converted model and then apply some numpy calculation & nms step to output bounding bo (followed #343 ) . But its seems NMS step doesn't working. Could you share your script to inference of converted openvino model ? if possile, could you give me your contact ? Thank you in advance !!!
> ![image](https://user-images.githubusercontent.com/30823943/90867641-e9dc2980-e3bf-11ea-9359-8b2c51a031ad.png)

ok,359684740@qq.com.

**Comment by linhaoqi027** [comment]
> @anhnktp can you tell me please if you have output with 5d shape like 1x80x80x3x7 cause i have an error when i run my transformed yolov5 openvino (outlayer ie::layout::NCDHW || outlayer ie::layout::NDHWC)

I can't known the reason without debug.But u can transform output to tensor then apply nms in this project.It works.

**Comment by samirMeziane** [comment]
thanks for your reply , my question was that i transformed onnx to xml format but when i try to do inference in openvino i get errors can you please tell me what wrong in my xml file 
[openvio(1).zip](https://github.com/ultralytics/yolov5/files/5149769/openvio.1.zip)

**Comment by linhaoqi027** [comment]
> thanks for your reply , my question was that i transformed onnx to xml format but when i try to do inference in openvino i get errors can you please tell me what wrong in my xml file
> [openvio(1).zip](https://github.com/ultralytics/yolov5/files/5149769/openvio.1.zip)
Oh I will upload my project soon, and you can read my code directly soon. Sorry I dont have enough time to debug your code.

**Comment by linhaoqi027** [comment]
> > 感谢您的答复，我的问题是我将onnx转换为xml格式，但是当我尝试在openvino中进行推断时遇到错误，您能告诉我xml文件
> > [openvio（1）.zip中有](https://github.com/ultralytics/yolov5/files/5149769/openvio.1.zip)什么问题[吗，](https://github.com/ultralytics/yolov5/files/5149769/openvio.1.zip) 
> > 哦，我将上传我的项目很快，您可以很快直接阅读我的代码。抱歉，我没有足够的时间来调试您的代码。

you can following https://github.com/ultralytics/yolov5/issues/891 to start

**Comment by glenn-jocher** [comment]
@linhaoqi027 @anhnktp @samirMeziane @Aktcob good news 😃! Your original issue may now be fixed ✅ in PR #6057. This PR adds native YOLOv5 OpenVINO export:

```bash
python export.py --weights yolov5s.pt --include openvino  # export to OpenVINO
```
![image](https://user-images.githubusercontent.com/26833433/147145516-b4f09a44-5679-472a-b512-97c51adffefc.png)


To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
